File: colaborandoBack/ColaborandoApplication/src/main/java/com/example/ColaborandoApplication/service/ReporteHorasService.java
package com.example.ColaborandoApplication.service;

import com.example.ColaborandoApplication.Entity.*;
import com.example.ColaborandoApplication.repository.ReporteHorasRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.util.List;


@Service
public class ReporteHorasService {

    @Autowired
    private ReporteHorasRepository reporteHorasRepository;



    public List<ReporteHoras> getHorasColaborador(Integer colaborador){
        try{
            List<ReporteHoras> horas = reporteHorasRepository.findByHorasColaborador(colaborador);
            return horas;
        } catch (Exception e) {
            System.out.println("Error al buscar los Eventos: {}"+ e.getMessage()+"\n");
        }
        return  null;
    }

    public List<ReporteHoras> getHorasEmpresa(){
        try{
            List<ReporteHoras> horas = reporteHorasRepository.findAll();
            return horas;
        } catch (Exception e) {
            System.out.println("Error al buscar los Eventos: {}"+ e.getMessage()+"\n");
        }
        return  null;
    }

}

File: colaborandoFront/src/services/ReporteHorasService.js
import axios from 'axios';

const EVENTO_API_BASE_URL = "http://colaborando.ddns.net:8080/colaborando/reporteHoras";

class ReporteHorasService { 

    getHorasColaborador(colaborador){
        return axios.get(`${EVENTO_API_BASE_URL}/horas-colaborador/${colaborador}`, {});
    }  

    getHorasEmpresa(){
        console.log(`${EVENTO_API_BASE_URL}/horas-empresa`);
        return axios.get(`${EVENTO_API_BASE_URL}/horas-empresa`);

    }      

}

const ReporteHorasServiceService = new ReporteHorasService();

export default ReporteHorasServiceService;